doc, ci: precise commit style

bougue-pe opened the pull request:
also port last check of commit after ~https://github.com/osrd-project/osrd/pull/6557~ https://github.com/osrd-project/osrd/pull/6580

TODO:
- [x] sync regex of this repository once `osrd` is merged
- [x] wait for merge of #154 and rebase + autosquash

Khoyo commented on content/docs/guides/contribute/code.en.md:
```diff
@@ -87,13 +87,31 @@ This chapter is about the process of integrating changes into the common code ba
+
+### Commit counter-examples
+
+- `component: update ./lol/mdr.rs`: specify the update itself rather than the file.
```
`some/file.rs`

flomonster reviewed (approved):
LGTM after fixing @Khoyo's comments

bougue-pe commented:
@Khoyo : rebased and corrected according to your remarks in https://github.com/osrd-project/osrd-website/pull/173/commits/02d58054763a1999f4da48e0157972554cc08aff (also synced after https://github.com/osrd-project/osrd/pull/6580)

All is autosquashed, you may merge if it's OK for you.
